packages/flutter/lib/src/material/slider.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'theme.dart';
+
+const double _kThumbRadius = 6.0;
+const double _kReactionRadius = 16.0;
+
+class Slider extends StatelessComponent {
+  Slider({ Key key, this.value, this.onChanged })
+      : super(key: key);
+
+  final double value;
+  final ValueChanged<double> onChanged;
+
+  Widget build(BuildContext context) {
+    return new _SliderWrapper(
+      value: value,
+      primaryColor: Theme.of(context).accentColor,
+      onChanged: onChanged
+    );
+  }
+}
+
+class _SliderWrapper extends LeafRenderObjectWidget {
+  _SliderWrapper({ Key key, this.value, this.primaryColor, this.onChanged })
+      : super(key: key);
+
+  final double value;
+  final Color primaryColor;
+  final ValueChanged<double> onChanged;
+
+  _RenderSlider createRenderObject() => new _RenderSlider(
+    value: value,
+    primaryColor: primaryColor,
+    onChanged: onChanged
+  );
+
+  void updateRenderObject(_RenderSlider renderObject, _SliderWrapper oldWidget) {
+    renderObject.value = value;
+    renderObject.primaryColor = primaryColor;
+    renderObject.onChanged = onChanged;
+  }
+}
+
+final Color _kInactiveTrackColor = Colors.grey[400];
+final Color _kActiveTrackColor = Colors.grey[500];
+
+class _RenderSlider extends RenderConstrainedBox {
+  _RenderSlider({
+    double value,
+    Color primaryColor,
+    this.onChanged
+  }) : _value = value,
+       _primaryColor = primaryColor,
+        super(additionalConstraints: const BoxConstraints.tightFor(width: 144.0, height: 2 * _kReactionRadius)) {
+    _drag = new HorizontalDragGestureRecognizer(router: FlutterBinding.instance.pointerRouter)
+      ..onStart = _handleDragStart
+      ..onUpdate = _handleDragUpdate
+      ..onEnd = _handleDragEnd;
+  }
+
+  double get value => _value;
+  double _value;
+  void set value(double newValue) {
+    if (newValue == _value)
+      return;
+    _value = newValue;
+    markNeedsPaint();
+  }
+
+  Color get primaryColor => _primaryColor;
+  Color _primaryColor;
+  void set primaryColor(Color value) {
+    if (value == _primaryColor)
+      return;
+    _primaryColor = value;
+    markNeedsPaint();
+  }
+
+  ValueChanged<double> onChanged;
+
+  HorizontalDragGestureRecognizer _drag;
+  bool _active = false;
+  double _currentDragValue = 0.0;
+
+  void _handleDragStart(Point globalPosition) {
+    _active = true;
+    markNeedsPaint();
+    if (onChanged != null) {
```

Comment:
you go active regardless of whether you have an onChanged, but you don't change unless you do. It seems like these should be consistent. Either we set _currentDragValue in all cases, or we use the lack of onChanged to disable ourselves entirely.